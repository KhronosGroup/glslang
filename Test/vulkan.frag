#version 450

uniform sampler s;
uniform sampler sA[4];
uniform texture2D t2d;
uniform texture3D t3d[4];
uniform int i;

sampler2D s2Db1 = sampler2D(t2d);       // ERROR, need 2 args
sampler2D s2Db2 = sampler2D(s, s);      // ERROR, wrong type
sampler2D s2Db3 = sampler2D(i, i);      // ERROR, wrong type
sampler2D s2Db4 = sampler2D(t2d, i);    // ERROR, wrong type
sampler2D s2Db5 = sampler2D(t2d, t2d);  // ERROR, wrong type
sampler2D s2Db6 = sampler2D(t2d, sA);   // ERROR, wrong type

sampler3D s3Db1 = sampler2D(t2d, s);              // ERROR, can't convert
uniform texture3D t3d5[5];
sampler3D s3d[4] = sampler3D[4](t3d5, sA[2]);     // ERROR, array size mismatch
writeonly uniform image2D i2d;
sampler2D s2Db7 = sampler2D(i2d, s);              // ERROR, image instead of texture
sampler2D s2Db8 = sampler2D(t3d[1], s);           // ERROR, 3D not 2D
sampler3D s3db1[4] = sampler3D[5](t3d5, sA[2]);   // ERROR, array size mismatch
sampler3D s3db2[4] = sampler3D[4](t3d[1], sA[2]); // ERROR, array/scalar mismatch

sampler2D s2D = sampler2D(t2d, s);
sampler3D s3d[4] = sampler3D[4](t3d, sA[2]);

out vec4 color;

void main()
{
    color = texture(s2D, vec2(0.5));
    color += texture(s3d[i], vec3(0.5));
}
