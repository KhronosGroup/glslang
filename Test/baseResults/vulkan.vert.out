../Test/vulkan.vert
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.
ERROR: ../Test/vulkan.vert:3: 'subpass input' : not supported in this stage: vertex
ERROR: ../Test/vulkan.vert:4: 'subpass input' : not supported in this stage: vertex
ERROR: ../Test/vulkan.vert:5: 'subpass input' : not supported in this stage: vertex
ERROR: ../Test/vulkan.vert:6: 'subpass input' : not supported in this stage: vertex
ERROR: ../Test/vulkan.vert:7: 'subpass input' : not supported in this stage: vertex
ERROR: ../Test/vulkan.vert:8: 'subpass input' : not supported in this stage: vertex
ERROR: ../Test/vulkan.vert:12: 'constant_id' : can only be applied to a scalar 
ERROR: ../Test/vulkan.vert:13: 'constant_id' : specialization-constant id already used 
ERROR: ../Test/vulkan.vert:13: 'constant_id' : can only be applied to 'const'-qualified scalar 
ERROR: ../Test/vulkan.vert:13: 'constant_id' : cannot be applied to this type 
ERROR: ../Test/vulkan.vert:14: 'constant_id' : specialization-constant id is too large 
ERROR: ../Test/vulkan.vert:15: 'constant_id' : can only be applied to a scalar 
ERROR: ../Test/vulkan.vert:16: 'constant_id' : specialization-constant id already used 
ERROR: ../Test/vulkan.vert:16: 'constant_id' : cannot declare a default, can only be used on a scalar 
ERROR: ../Test/vulkan.vert:20: 'subpassLoad' : no matching overloaded function found 
ERROR: ../Test/vulkan.vert:20: 'assign' :  cannot convert from ' const float' to ' smooth out highp 4-component vector of float'
ERROR: ../Test/vulkan.vert:23: 'atomic counter types' : not allowed when using GLSL for Vulkan 
ERROR: ../Test/vulkan.vert:24: 'shared' : not allowed when generating SPIR-V 
ERROR: ../Test/vulkan.vert:25: 'packed' : not allowed when generating SPIR-V 
ERROR: ../Test/vulkan.vert:32: 'initializer' : can't use with types containing arrays sized with a specialization constant 
ERROR: ../Test/vulkan.vert:34: '=' : can't use with types containing arrays sized with a specialization constant 
ERROR: ../Test/vulkan.vert:35: '==' : can't use with types containing arrays sized with a specialization constant 
ERROR: ../Test/vulkan.vert:39: 'set' : cannot be used with push_constant 
ERROR: 23 compilation errors.  No code generated.


SPIR-V is not generated for failed compile or link
